### DISCUSSION TITLE
Empty strings returned from valid mathematical operations

### DISCUSSION THREAD

**Discussion by marcuskamkpmg** [sinaptik-ai/pandas-ai#325] [Q&A]
Hello all, 

I've been running into this problem where pandas_ai returns the average of a column but not the sum of the column. The logs are shown below where 'July Budget' is a column containing only floats:

df['July Budget'].mean() <- this returns a string that contains the correct value 
df['July Budget'].sum() <- this returns an empty string

Any idea why this happened? Thanks so much!

**Comment by gventuri** [comment]
Hey @marcuskamkpmg could you share the `pandasai.log` file or some more info? The code generated looks good to me, but I'm wondering whether it's the whole code or there's something more.

**Reply by marcuskamkpmg** [reply]
Apologies, but I can't seem to replicate the log output anymore. Anyhow, I noticed the model sometimes switches between the df[df[]] method and df.loc method, which could have produced inconsistent results. Moreover, the model mislabelled the searching columns at times (had ['Yearly Budget'] when asked explicitly for ['July budget']).

**Reply by gventuri** [reply]
@marcuskamkpmg So the column name was 'July budget' and the LLM generated 'July Budget'?

**Reply by marcuskamkpmg** [reply]
Yes, upper/lower case errors were present. Moreover, sometimes it even generates the column label with a space (df['Business Code']) when there shouldn't be (it's df['BusinessCode'] in the dataframe), or just have the wrong column label to start with.
